src/App/Entity/Meetup.php

```diff
@@ -3,201 +3,134 @@
      * @var DateTimeImmutable
      */
     private $toDate;
 
     /**
-     * @var string
+     * @ORM\OneToOne(targetEntity=EventbriteData::class, mappedBy="meetup")
+     * @var EventbriteData
      */
-    private $registrationUrl;
+    private $eventbriteData;
 
     /**
-     * @var string
-     */
-    private $locationUrl;
-
-    /**
-     * @var string
+     * @ORM\ManyToOne(targetEntity=Location::class)
+     * @ORM\JoinColumn(name="location_id", referencedColumnName="id", nullable=false)
+     * @var Location
      */
     private $location;
 
     /**
+     * @ORM\Column(name="topic", type="string", length=1024, nullable=true)
      * @var string
      */
     private $topic;
 
     /**
-     * @var Talk[]|string[]
-     */
-    private $talkingPoints = [];
-
-    /**
-     * @var string
+     * @ORM\OneToMany(targetEntity=Talk::class, mappedBy="meetup")
+     * @ORM\OrderBy({"time" = "ASC"})
+     * @var ArrayCollection|Talk[]
      */
-    private $widget;
-
-    /**
-     * @var Schedule[]
-     */
-    private $schedule = [];
-
-    public function __construct()
-    {
-        $this->schedule = [];
-    }
+    private $talks;
 
-    public function exchangeArray($data) : array
+    private function __construct()
     {
-        throw new \Exception('Not implemented yet...');
-    }
-
-    public function setId(int $id) : self
-    {
-        $this->id = $id;
-
-        return $this;
+        $this->id = Uuid::uuid4();
+        $this->talks = new ArrayCollection();
     }
 
     /**
-     * @return int
+     * @param DateTimeImmutable $from
+     * @param DateTimeImmutable $to
+     * @param Location $location
+     * @param Talk[] $talks
+     * @param string $topic
+     * @return Meetup
+     * @throws \InvalidArgumentException
      */
-    public function getId() : int
-    {
-        return $this->id;
-    }
-
-    public function setFromDate(DateTimeImmutable $date) : self
-    {
-        $this->fromDate = $date;
+    public static function fromStandardMeetup(
+        DateTimeImmutable $from,
+        DateTimeImmutable $to,
+        Location $location,
+        array $talks,
+        string $topic = null
+    ) : self {
+        $meetup = new self();
+        $meetup->fromDate = $from;
+        $meetup->toDate = $to;
+        $meetup->location = $location;
+        $meetup->topic = $topic;
 
-        return $this;
+        foreach ($talks as $k => $talk) {
+            if (!$talk instanceof Talk) {
+                throw new \InvalidArgumentException(sprintf('Item with key %s in talks was not a Talk', $k));
+            }
+            $meetup->talks->add($talk);
+        }
+        return $meetup;
     }
 
-    /**
-     * @return DateTimeImmutable
-     */
-    public function getFromDate() : DateTimeImmutable
+    public function getFromDate() : \DateTimeImmutable
     {
-        return $this->fromDate;
+        return new \DateTimeImmutable($this->fromDate->format('Y-m-d H:i:s'));
```

Comment:
I think I sort of understand, but it seems a bit excessive to have to type out the format string each time